pipeline/daily/stages.groovy

```diff
@@ -108,4 +109,35 @@ def createSymlinks() {
   utils.rsyncUpload('--links latest', PERIODIC_BUILDS_DIR_RSYNC_URL)
 }
 
+def shouldNotifyOnFailure() {
+  shouldNotify = true
+
+  for (param in params) {
+    if (param.key.startsWith("SLACK") && !param.value) {
+      shouldNotify = false
+      break
```

Comment:
You could just `return false` here and `return true` outside the for, removing this variable. Up to you.